integration/spark-common-cluster-test/src/test/scala/org/apache/spark/sql/common/util/QueryTest.scala

```diff
@@ -84,22 +82,34 @@ class QueryTest extends PlanTest with Suite {
     checkAnswer(df, expectedAnswer.collect())
   }
 
-  protected def checkAnswer(carbon: String, hive: String, uniqueIdentifier:String): Unit = {
-    val path = TestQueryExecutor.hiveresultpath + "/"+uniqueIdentifier
+  protected def checkAnswer(carbon: String, hive: String, uniqueIdentifier: String): Unit = {
+    val path = TestQueryExecutor.hiveresultpath + "/" + uniqueIdentifier
     if (FileFactory.isFileExist(path, FileFactory.getFileType(path))) {
-      val objinp = new ObjectInputStream(FileFactory.getDataInputStream(path, FileFactory.getFileType(path)))
+      val objinp = new ObjectInputStream(FileFactory
+        .getDataInputStream(path, FileFactory.getFileType(path)))
       val rows = objinp.readObject().asInstanceOf[Array[Row]]
       objinp.close()
-      checkAnswer(sql(carbon), rows)
+      QueryTest.checkAnswer(sql(carbon), rows) match {
+        case Some(errorMessage) => {
+          FileFactory.deleteFile(path, FileFactory.getFileType(path))
+          writeAndCheckAnswer(carbon, hive, path)
```

Comment:
Doesn't it go to endless loop when test fails?

Reply:
i couldn't understand your comment. how it would go to infinite loop?
we are not using recursive call

Reply:
Got it , my misunderstanding